src/Rest/Routes/NewsRoute.php

```diff
@@ -0,0 +1,140 @@
+use UnicornsVendor\EightshiftLibs\Rest\Routes\AbstractRoute;
+use UnicornsVendor\EightshiftLibs\Rest\CallableRouteInterface;
+
+/**
+ * Class NewsRoute
+ */
+class NewsRoute extends AbstractRoute implements CallableRouteInterface
+{
+
+	/**
+	 * Method that returns project Route namespace.
+	 *
+	 * @return string Project namespace UnicornsVendor\for REST route.
+	 */
+	protected function getNamespace(): string
+	{
+		return Config::getProjectRoutesNamespace();
+	}
+
+	/**
+	 * Method that returns project route version.
+	 *
+	 * @return string Route version as a string.
+	 */
+	protected function getVersion(): string
+	{
+		return Config::getProjectRoutesVersion();
+	}
+
+	/**
+	 * Get the base url of the route
+	 *
+	 * @return string The base URL for route you are adding.
+	 */
+	protected function getRouteName(): string
+	{
+		return '/news';
+	}
+
+	/**
+	 * Get callback arguments array
+	 *
+	 * @return array Either an array of options for the endpoint, or an array of arrays for multiple methods.
+	 */
+	protected function getCallbackArguments(): array
+	{
+		return [
+			'methods' => static::READABLE,
+			'callback' => [$this, 'routeCallback'],
+			'permission_callback' => '__return_true'
+		];
+	}
+
+	/**
+	 * Method that returns rest response
+	 *
+	 * @param \WP_REST_Request $request Data got from endpoint url.
+	 *
+	 * @return \WP_REST_Response|mixed If response generated an error, WP_Error, if response
+	 *                                is already an instance, WP_HTTP_Response, otherwise
+	 *                                returns a new WP_REST_Response instance.
+	 */
+	public function routeCallback(\WP_REST_Request $data)
+	{
+		$params = $data->get_params();
+
+		$returnHtml = false;
+
+		foreach ($params as $name => $value) {
+			if ($name != '_limit' && $name != '_start') {
+				unset($params[$name]);
+			}
+			if ($name == 'html') {
+				$returnHtml = true;
+			}
+		}
+
+		$parsedParams = "";
+
+		if (sizeof($params) > 0) {
+			$parsedParams = '?' . http_build_query($params);
+		}
+
+		$url = "https://test.spaceflightnewsapi.net/api/v2/articles$parsedParams";
+		$remoteResponse = \wp_remote_get($url);
+
+		if (is_array($remoteResponse)) {
+			$body = json_decode($remoteResponse['body'], true);
+		}
+
+		if (!isset($body)) {
+			return new \WP_Error('no_news', 'there are no news available', ['status' => 404]);
+		}
+
+		if ($returnHtml) {
+			$outputHtml = '';
+
+			foreach ($body as $post) {
+				$postImage = $post['imageUrl'];
+
+				$postDate = date_create($post['publishedAt']);
+
+				$postTitle = $post['title'];
+
+				$postExcerpt = $post['summary'];
+
+				$postCardProps = [
+					'imageUrl' => wp_kses_post($postImage),
+					'imageUse' => $postImage ?? true,
+					'dateContent' => date_format($postDate, 'j.m.Y @ G:i'),
+					'headingContent' => wp_kses_post($postTitle),
+					'excerptContent' => wp_kses_post($postExcerpt),
+					'tagsContent' => '<li>Prvo</li><li>Drugo</li><li>Trece</li>'
```

Comment:
what?

Reply:
The idea was to escape the content loaded from the API before passing it on to the Card element, but wasn't really sure which function to use for that.

Is `esc_url_raw` for the image url and `esc_attr` for heading, excerpt, ... ok or should I use something else?

Tags I hardcoded here because the API didn't have them 😅

Reply:
un general we almost never use esc_url_raw. And escaping is that you did is ok.

The key name string here is 'hardcoded' i don't want to se hardcoded any where. If the component doesn't have the tags don't show them